src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1646,6 +1646,40 @@ private IForLoopOperation CreateBoundForStatementOperation(BoundForStatement bou
                 HashSet<DiagnosticInfo>? useSiteDiagnostics = null;
                 var compilation = (CSharpCompilation)_semanticModel.Compilation;
 
+                ImmutableArray<IArgumentOperation> getEnumeratorArguments;
+                if (enumeratorInfoOpt.GetEnumeratorMethod is { IsExtensionMethod: true } enumeratorMethod)
```

Comment:
>true [](start = 82, length = 4)

Can we run into an instance method with default values and/or param array?

Reply:
This commit is no longer necessary, dropping from the updated PR.

---
In reply to: [529836543](https://github.com/dotnet/roslyn/pull/49534#discussion_r529836543) [](ancestors = 529836543)